File: app/src/main/java/top/sacz/hook/HookSteps.java
package top.sacz.hook;

import android.content.Context;
import android.content.ContextWrapper;

import java.lang.reflect.Method;

import de.robv.android.xposed.callbacks.XC_LoadPackage;

public class HookSteps {

    /**
     * 获取初始的Hook方法
     *
     * @param loadPackageParam
     * @return
     */
    public Method getApplicationCreateMethod(XC_LoadPackage.LoadPackageParam loadPackageParam) {
        try {
            String applicationName = loadPackageParam.appInfo.name;
            Class<?> clz = loadPackageParam.classLoader.loadClass(applicationName);
            try {
                return clz.getDeclaredMethod("attachBaseContext", Context.class);
            } catch (Throwable i) {
                try {
                    return clz.getDeclaredMethod("onCreate");
                } catch (Throwable e) {
                    try {
                        return clz.getSuperclass().getDeclaredMethod("attachBaseContext", Context.class);
                    } catch (Throwable m) {
                        return clz.getSuperclass().getDeclaredMethod("onCreate");
                    }
                }
            }
        } catch (Exception e) {
            try {
                return ContextWrapper.class.getDeclaredMethod("attachBaseContext", Context.class);
            } catch (NoSuchMethodException ex) {
                return null;
            }
        }
    }

    public void initHook(Context context) {
        new DexkitTest().hook();
    }
}

File: sample/src/main/java/top/sacz/hook/HookSteps.kt
package top.sacz.hook

import android.content.Context
import android.content.ContextWrapper
import de.robv.android.xposed.callbacks.XC_LoadPackage.LoadPackageParam
import java.lang.reflect.Method

class HookSteps {
    /**
     * 获取初始的Hook方法
     *
     * @param loadPackageParam
     * @return
     */
    fun getApplicationCreateMethod(loadPackageParam: LoadPackageParam): Method? {
        try {
            val applicationName = loadPackageParam.appInfo.name
            val clz = loadPackageParam.classLoader.loadClass(applicationName)
            return try {
                clz.getDeclaredMethod("attachBaseContext", Context::class.java)
            } catch (i: Throwable) {
                try {
                    clz.getDeclaredMethod("onCreate")
                } catch (e: Throwable) {
                    try {
                        clz.getSuperclass()
                            .getDeclaredMethod("attachBaseContext", Context::class.java)
                    } catch (m: Throwable) {
                        clz.getSuperclass().getDeclaredMethod("onCreate")
                    }
                }
            }
        } catch (e: Exception) {
            return try {
                ContextWrapper::class.java.getDeclaredMethod(
                    "attachBaseContext",
                    Context::class.java
                )
            } catch (ex: NoSuchMethodException) {
                null
            }
        }
    }
    fun initHook(context: Context?) {
        DexkitTest().hook()
    }
}